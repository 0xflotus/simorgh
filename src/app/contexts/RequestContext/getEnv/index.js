const getEnv = origin => {
  if (
    origin.includes('local') ||
    (process && process.env && process.env.APP_ENV === 'local')
  ) {
    return 'local';
  }

  if (origin.includes('test')) {
    return 'test';
  }

  return 'live';
};

export default getEnv;
