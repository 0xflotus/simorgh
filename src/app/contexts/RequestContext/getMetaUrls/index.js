const base = `https://www.`

const getAmpUrl = url => `${url}.amp`

const getCanonicalUrl = (origin, service, pageType, id) => {
    switch (pageType) {
        case 'frontPage':
            return `${origin}/${service}`;
        default:
            return `${origin}/${service}/articles/${id}`;
    }
}

const getCanonicalUrlWithVariant = (origin, service, serviceVariant, pageType, id) => {
    switch (pageType) {
        case 'frontPage':
            return `${origin}/${service}/${serviceVariant}`;
        default:
            return `${origin}/${service}/articles/${id}/${serviceVariant}`;
    }
}

const getUkConicalUrl = (url) => {
    const tldRegex = /(.com|.co.uk)/g
    return url.replace(tldRegex, '.com')
}

const getNonUkConicalUrl = (url) => {
    const tldRegex = /(.com|.co.uk)/g
    return url.replace(tldRegex, '.co.uk')
}

const getMetaUrls = (origin, env, platform, service, serviceVariant, pageType, id) => {
    let
        canonicalLink = getCanonicalUrl(origin, service, pageType, id),
        canonicalUkLink = getUkConicalUrl(canonicalLink),
        canonicalNonUkLink = getNonUkConicalUrl(canonicalLink);

    if (serviceVariant) {
        canonicalLink = getCanonicalUrlWithVariant(origin, service, serviceVariant, pageType, id);
        return {
            canonicalLink: canonicalLink,
            ampLink: getAmpUrl(canonicalLink),
            canonicalUkLink: canonicalUkLink,
            ampUkLink: getAmpUrl(canonicalUkLink),
            canonicalNonUkLink: canonicalNonUkLink,
            ampNonUkLink: getAmpUrl(canonicalNonUkLink),
        }
    }

    return {
        canonicalLink: canonicalLink,
        ampLink: getAmpUrl(canonicalLink),
        canonicalUkLink: canonicalUkLink,
        ampUkLink: getAmpUrl(canonicalUkLink),
        canonicalNonUkLink: canonicalNonUkLink,
        ampNonUkLink: getAmpUrl(canonicalNonUkLink),
    }
}

export default getMetaUrls