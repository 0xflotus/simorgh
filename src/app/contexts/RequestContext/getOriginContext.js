import get from '../../helpers/json/deepGet';
import onClient from '../../helpers/onClient';

const getArticleID = articleData => {
  const aresID = get(['metadata', 'id'], articleData);
  return aresID ? aresID.split(':').pop() : 'unknown';
};

const getOrigin = bbcOrigin => {
  let origin = 'https://www.bbc.co.uk';

  if (bbcOrigin) {
    origin = bbcOrigin;
  } else if (onClient() && window.location.origin) {
    origin = window.location.origin; // eslint-disable-line prefer-destructuring
  }

  return origin;
};

const getOriginContext = (bbcOrigin, service, articleData) => {
  const origin = getOrigin(bbcOrigin);
  let href = `${origin}/${service}/${getArticleID(articleData)}`;
  let isUK = true;
  let referrer = null;
  let env = 'live';

  if (onClient()) {
    if (window.location.href) {
      href = window.location.href; // eslint-disable-line prefer-destructuring
    }

    if (document.referrer) {
      referrer = document.referrer; // eslint-disable-line prefer-destructuring
    }
  }

  if (origin.includes('.stage.')) {
    env = 'stage';
  }

  if (origin.includes('.test.')) {
    env = 'test';
  }

  if (origin.includes('localhost')) {
    env = 'local';
  }

  if (origin.includes('.com')) {
    isUK = false;
  }

  return {
    origin,
    isUK,
    env,
    href,
    referrer,
  };
};

export default getOriginContext;
