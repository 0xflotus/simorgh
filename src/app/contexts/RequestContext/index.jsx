import React from 'react';
import { bool, node, string } from 'prop-types';

export const RequestContext = React.createContext('default');

export const RequestContextProvider = ({
  children,
  env,
  id,
  isUK,
  origin,
  platform,
  statsDestination,
  statsPageIdentifier,
}) => {
  const value = {
    env,
    id,
    isUK,
    origin,
    platform,
    statsDestination,
    statsPageIdentifier,
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  children: node.isRequired,
  env: string,
  id: string,
  platform: string.isRequired,
  isUK: bool.isRequired,
  origin: string.isRequired,
  statsDestination: string.isRequired,
  statsPageIdentifier: string.isRequired,
};

RequestContextProvider.defaultProps = {
  id: null,
  env: 'local',
};
