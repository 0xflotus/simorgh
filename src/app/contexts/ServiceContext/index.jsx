import React from 'react';
import { node, string } from 'prop-types';
import services from '../../lib/config/services';
import defaultService from '../../lib/config/services/default';
import createLoadableContext from '../utils/createLoadableContext';

/* Create ServiceContext using the default service */
export const ServiceContext = React.createContext(defaultService);

/*
*
* It's important to note that this requires that you declare all of your loadable
* components when modules are initialized rather than when your app is being rendered.
* https://github.com/jamiebuilds/react-loadable#loadablepreloadall
* 
*/
const loadableContexts = {};

Object.keys(services).map(service => {
  loadableContexts[service] = createLoadableContext(
    ServiceContext,
    services[service],
  );
});

export const ServiceContextProvider = ({ children, service }) => {
  const LoadableServiceContextProvider = loadableContexts[service];

  if (!LoadableServiceContextProvider) {
    return null;
  }

  return (
    <LoadableServiceContextProvider>{children}</LoadableServiceContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
};
