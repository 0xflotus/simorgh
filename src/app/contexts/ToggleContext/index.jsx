import React, { createContext, useEffect, useReducer } from 'react';
import { node, string } from 'prop-types';
import { toggleReducer, updateToggles } from './reducer';
import defaultToggles from '#lib/config/toggles';
import onClient from '../../lib/utilities/onClient';

const ToggleContext = createContext({});

const getTogglesEndpoint = (service, origin) => {
  const requestOrigin = origin ? origin : 'https://www.test.bbc.com';
  const baseTogglesUrl = `${process.env.SIMORGH_TOGGLES_URL}/toggles?application=simorgh&service=${service}&__amp_source_origin=${requestOrigin}`;

  // client side renders should trigger a geoIPLookup
  if (onClient()) {
    return `${baseTogglesUrl}&geoiplookup=true`;
  }

  return baseTogglesUrl;
};

const ToggleContextProvider = ({ children, service, origin }) => {
  const simorghToggles = defaultToggles[process.env.SIMORGH_APP_ENV];
  const [toggleState, toggleDispatch] = useReducer(
    toggleReducer,
    simorghToggles,
  );

  // temp method to only enable remote freature toggling for test and for a list of services
  const { remoteFeatureToggles } = defaultToggles[process.env.SIMORGH_APP_ENV];

  if (
    remoteFeatureToggles.enabled &&
    service.match(remoteFeatureToggles.value)
  ) {
    useEffect(() => {
      const fetchAndUpdateToggles = async () => {
        const response = await fetch(getTogglesEndpoint(service, origin));
        const jsonData = await response.json();

        // container code: const { ads } = toggleContext(); if(ads && ads.enabled)
        // When we make the server request, the geoiplookup won't need to be made.
        // Containers that require a geoip-specific setup
        //
        toggleDispatch(updateToggles(jsonData));
      };
      fetchAndUpdateToggles();
    }, [service]);
  }

  return (
    <ToggleContext.Provider value={{ toggleState, toggleDispatch }}>
      {children}
    </ToggleContext.Provider>
  );
};

const ToggleContextConsumer = ToggleContext.Consumer;

ToggleContextProvider.propTypes = {
  children: node.isRequired,
  service: string.isRequired,
};

export { ToggleContext, ToggleContextProvider, ToggleContextConsumer };
