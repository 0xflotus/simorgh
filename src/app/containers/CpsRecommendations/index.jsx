import React, { useContext } from 'react';
import { arrayOf, shape, number, string } from 'prop-types';
import styled from 'styled-components';
import path from 'ramda/src/path';
import SectionLabel from '@bbc/psammead-section-label';
import StoryPromo, { Headline, Link } from '@bbc/psammead-story-promo';
import { StoryPromoLi, StoryPromoUl } from '@bbc/psammead-story-promo-list';
import { C_LUNAR, C_EBON, C_GHOST } from '@bbc/psammead-styles/colours';
import Image from '@bbc/psammead-image';
import { getDoublePica } from '@bbc/gel-foundations/typography';
import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import {
  GEL_SPACING,
  GEL_SPACING_HLF,
  GEL_SPACING_DBL,
} from '@bbc/gel-foundations/spacings';

import { ServiceContext } from '#contexts/ServiceContext';
import useToggle from '#hooks/useToggle';
import CpsOnwardJourney from '../CpsOnwardJourney';
import Grid from '../../components/Grid';
import data from './recommendations.ltr.json';

const StoryPromoWrapper = styled(StoryPromo)`
  display: grid;
  margin: ${GEL_SPACING_HLF} ${GEL_SPACING_DBL};
  background-color: ${C_GHOST};
`;

const CustomSectionLabel = styled(SectionLabel).attrs(props => ({
  'aria-labelledby': props.labelId,
}))`
  ${({ script }) => script && getDoublePica(script)};
  ${({ service }) => getSansRegular(service)};
  color: ${C_EBON};
  margin-top: 0;
  background-color: ${C_LUNAR};
`;

const SectionLabelWrapper = styled.div`
  padding: ${GEL_SPACING_DBL};
`;

const WsojWrapper = styled.div`
  background-color: ${C_LUNAR};
  padding-bottom: ${GEL_SPACING};
`;

const StyledStoryPromoLi = styled(StoryPromoLi)`
  /* Override various paddings set by StoryPromoLi */
  padding: 0 !important;
`;

const getPromoItemProps = item => ({
  src: path(['indexImage', 'path'], item),
  href: path(['locators', 'assetUri'], item),
  headline: path(['headlines', 'headline'], item),
});

const toStoryPromoItem = ({ assetUri, shortHeadline, imageHref }) => ({
  headlines: {
    headline: shortHeadline,
  },
  locators: {
    assetUri,
  },
  indexImage: {
    path: imageHref,
  },
  uri: assetUri,
});

const transformItems = items => items.map(toStoryPromoItem);

const CpsRecommendations = ({ parentColumns, items }) => {
  const { script, service, dir, recommendations } = useContext(ServiceContext);
  const { enabled } = useToggle('cpsRecommendations');

  const { hasStoryRecommendations } = recommendations;

  if (!hasStoryRecommendations || !enabled) return null;

  const singleTransform = item => {
    const { src, href, headline } = getPromoItemProps(item);

    const Img = <Image alt="" src={src} />;

    const Info = (
      <Headline script={script} service={service} promoHasImage>
        <Link href={href}>{headline}</Link>
      </Headline>
    );

    return (
      <Grid
        columns={{
          group0: 1,
          group1: 1,
          group2: 1,
          group3: 1,
          group4: 1,
          group5: 1,
        }}
        enableGelGutters
        dir={dir}
      >
        <StoryPromoWrapper image={Img} info={Info} />
      </Grid>
    );
  };

  const listTransform = promoItems => (
    <Grid
      columns={{
        group0: 6,
        group1: 6,
        group2: 6,
        group3: 6,
        group4: 6,
        group5: 6,
      }}
      as={StoryPromoUl}
      enableGelGutters
      dir={dir}
    >
      {promoItems.map(item => (
        <Grid
          item
          columns={{
            group0: 6,
            group1: 6,
            group2: 6,
            group3: 6,
            group4: 12,
            group5: 12,
          }}
          as={StyledStoryPromoLi}
          key={item.uri}
          dir={dir}
        >
          {singleTransform(item)}
        </Grid>
      ))}
    </Grid>
  );

  const renderSectionLabel = () => (
    <SectionLabelWrapper>
      <CustomSectionLabel
        script={script}
        service={service}
        dir={dir}
        labelId="recommendations-heading"
        bar={false}
        as="strong"
      >
        Recommendations
      </CustomSectionLabel>
    </SectionLabelWrapper>
  );

  return (
    <WsojWrapper>
      <CpsOnwardJourney
        labelId="recommendations-heading"
        title="Recommendations"
        content={transformItems(items)}
        parentColumns={parentColumns}
        singleTransform={singleTransform}
        listTransform={listTransform}
        renderCustomLabel={renderSectionLabel}
        hasCustomLabel
      />
    </WsojWrapper>
  );
};

export default CpsRecommendations;

CpsRecommendations.propTypes = {
  items: arrayOf(
    shape({
      assetUri: string.isRequired,
      shortHeadline: string.isRequired,
      imageHref: string.isRequired,
    }),
  ),
  parentColumns: shape({
    group0: number,
    group1: number,
    group2: number,
    group3: number,
    group4: number,
    group5: number,
  }).isRequired,
};

CpsRecommendations.defaultProps = {
  items: data.items,
};
