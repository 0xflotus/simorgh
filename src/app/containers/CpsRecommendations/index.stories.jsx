import React from 'react';
import { storiesOf } from '@storybook/react';

import CpsRecommendations from '.';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ToggleContext } from '#contexts/ToggleContext';

import mundoRecommendationsData from './recommendations.ltr.json';
import arabicRecommendationsData from './recommendations.rtl.json';

const renderRecommendations = (items, service) => {
  const toggleState = {
    cpsRecommendations: {
      enabled: true,
    },
  };
  const parentGridCols = {
    group0: 8,
    group1: 8,
    group2: 8,
    group3: 8,
    group4: 8,
    group5: 8,
  };
  return (
    <ServiceContextProvider service={service}>
      <RequestContextProvider
        isAmp={false}
        pageType="STY"
        service={service}
        pathname="/pathname"
      >
        <ToggleContext.Provider
          value={{ toggleState, toggleDispatch: () => {} }}
        >
          <CpsRecommendations items={items} parentColumns={parentGridCols} />
        </ToggleContext.Provider>
      </RequestContextProvider>
    </ServiceContextProvider>
  );
};

storiesOf('Containers|CPS Recommendations', module)
  .addParameters({ chromatic: { disable: true } })
  .add('mundo (ltr)', () =>
    renderRecommendations(mundoRecommendationsData.items, 'mundo'),
  )
  .add('arabic (ltr)', () =>
    renderRecommendations(arabicRecommendationsData.items, 'arabic'),
  );
