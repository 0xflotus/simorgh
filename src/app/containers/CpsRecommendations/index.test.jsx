import React from 'react';
import { render } from '@testing-library/react';
import { latinDiacritics } from '@bbc/gel-foundations/scripts';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ToggleContext } from '#contexts/ToggleContext';
import recommendationData from './recommendations.ltr.json';

import CpsRecommendations from '.';

const singleRecommendationData = [
  {
    assetUri: '/mundo/5331',
    shortHeadline: 'This is a headline',
    imageHref: 'http://c.files.bbci.co.uk/256/727445.jpg',
  },
];

const multipleRecommendationData = recommendationData.items;

const parentGridCols = {
  group0: 8,
  group1: 8,
  group2: 8,
  group3: 8,
  group4: 8,
  group5: 8,
};

const renderContainer = (items, hasStoryRecommendations, toggleEnabled) => {
  const toggleState = {
    cpsRecommendations: {
      enabled: toggleEnabled,
    },
  };
  return render(
    <ServiceContext.Provider
      value={{
        script: latinDiacritics,
        service: 'mundo',
        dir: 'ltr',
        recommendations: {
          hasStoryRecommendations,
        },
      }}
    >
      <RequestContextProvider
        bbcOrigin="https://www.test.bbc.co.uk"
        isAmp={false}
        pageType="STY"
        pathname="/mundo/085965"
        service="mundo"
        statusCode={200}
      >
        <ToggleContext.Provider
          value={{ toggleState, toggleDispatch: jest.fn() }}
        >
          <CpsRecommendations items={items} parentColumns={parentGridCols} />
        </ToggleContext.Provider>
      </RequestContextProvider>
    </ServiceContext.Provider>,
  );
};

describe('CpsRecommendations', () => {
  it('should not render when cpsRecommendations toggle is disabled', () => {
    const toggleEnabled = false;
    const hasStoryRecommendations = true;
    const { container } = renderContainer(
      multipleRecommendationData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });
  it('should not render when the hasStoryRecommendations flag is disabled for the service', () => {
    const hasStoryRecommendations = false;
    const toggleEnabled = true;
    const { container } = renderContainer(
      multipleRecommendationData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render for multiple recommendation items', () => {
    const toggleEnabled = true;
    const hasStoryRecommendations = true;
    const { container } = renderContainer(
      multipleRecommendationData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render for a single recommendation item', () => {
    const toggleEnabled = true;
    const hasStoryRecommendations = true;
    const { container } = renderContainer(
      singleRecommendationData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });
});
