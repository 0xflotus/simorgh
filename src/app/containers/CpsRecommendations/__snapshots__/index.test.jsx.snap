// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpsRecommendations should not render when cpsRecommendations toggle is disabled 1`] = `<div />`;

exports[`CpsRecommendations should not render when the hasStoryRecommendations flag is disabled for the service 1`] = `<div />`;

exports[`CpsRecommendations should render for a single recommendation item 1`] = `
<div>
  <div
    class="WsojWrapper-sc-1p91yk0-3 bybDJb"
  >
    <section
      aria-labelledby="recommendations-heading"
      class="Wrapper-opyob2-0 dlivvF"
      data-e2e="related-content"
      role="region"
    >
      <div
        class="SectionLabelWrapper-sc-1p91yk0-2 keYZPs"
      >
        <strong
          class="CustomSectionLabel-sc-1p91yk0-1 dbbYpP"
          dir="ltr"
          id="recommendations-heading"
        >
          Recommendations
        </strong>
      </div>
      <div
        class="SingleContentWrapper-opyob2-3 iddZTf"
      >
        <div
          class="GridComponent-nf79gm-0 elxESl"
          dir="ltr"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 gsqqaw"
          >
            <div
              class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
              dir="ltr"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
              >
                <img
                  alt="Image Alt text"
                  class="StyledImg-sc-7vx2mr-0 hIxkbt"
                  src="http://c.files.bbci.co.uk/256/727445.jpg"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1lkbq5i-0 jJWtYv"
              dir="ltr"
            >
              <span
                class="Headline-sc-1dvfmi3-3 dnWhDh"
              >
                <a
                  class="Link-sc-1dvfmi3-5 jeUMCT"
                  href="/mundo/5331"
                >
                  This is a headline
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`CpsRecommendations should render for multiple recommendation items 1`] = `
<div>
  <div
    class="WsojWrapper-sc-1p91yk0-3 bybDJb"
  >
    <section
      aria-labelledby="recommendations-heading"
      class="Wrapper-opyob2-0 dlivvF"
      data-e2e="related-content"
      role="region"
    >
      <div
        class="SectionLabelWrapper-sc-1p91yk0-2 keYZPs"
      >
        <strong
          class="CustomSectionLabel-sc-1p91yk0-1 dbbYpP"
          dir="ltr"
          id="recommendations-heading"
        >
          Recommendations
        </strong>
      </div>
      <ul
        class="StoryPromoUl-sc-171lqjd-1 isCYWc GridComponent-nf79gm-0 buIDdq"
        dir="ltr"
        role="list"
      >
        <li
          class="StoryPromoLi-sc-171lqjd-0 StyledStoryPromoLi-sc-1p91yk0-4 cFGyOa GridComponent-nf79gm-0 dbJPTO"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 ehumcZ"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 gsqqaw"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Image Alt text"
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="http://c.files.bbci.co.uk/256/727445.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jJWtYv"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/5331"
                  >
                    This is a headline
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLi-sc-171lqjd-0 StyledStoryPromoLi-sc-1p91yk0-4 cFGyOa GridComponent-nf79gm-0 dbJPTO"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 ehumcZ"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 gsqqaw"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Image Alt text"
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="http://c.files.bbci.co.uk/123/af07635.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jJWtYv"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/12789"
                  >
                    This is another headline
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
